frontend/src/metabase/components/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,274 @@
+  scheduleOptions: ScheduleType[];
+  timezone?: string;
+  verb?: string;
+  textBeforeSendTime?: string;
+  minutesOnHourPicker?: boolean;
+  onScheduleChange: (
+    nextSchedule: ScheduleSettings,
+    change: ScheduleChangeProp,
+  ) => void;
+}) => {
+  const updateSchedule: UpdateSchedule = useCallback(
+    (field: ScheduleProperty, value: ScheduleSettings[typeof field]) => {
+      let newSchedule: ScheduleSettings = {
+        ...schedule,
+        [field]: value,
+      };
+
+      newSchedule = pick(newSchedule, val => val);
+
+      if (field === "schedule_type") {
+        newSchedule = {
+          ...defaults[value as ScheduleType],
+          ...newSchedule,
+        };
+      } else if (field === "schedule_frame") {
+        // when the monthly schedule frame is the 15th, clear out the schedule_day
+        if (value === "mid") {
+          newSchedule = { ...newSchedule, schedule_day: null };
+        } else {
+          // first or last, needs a day of the week
+          newSchedule = {
+            schedule_day: newSchedule.schedule_day || defaultDay,
+            ...newSchedule,
+          };
+        }
+      }
+
+      onScheduleChange(newSchedule, { name: field, value });
+    },
+    [onScheduleChange, schedule],
+  );
+
+  return (
+    <Box lh="41px" display="flex" style={{ flexWrap: "wrap", gap: ".5rem" }}>
+      <ScheduleBody
+        schedule={schedule}
+        updateSchedule={updateSchedule}
+        scheduleOptions={scheduleOptions}
+        timezone={timezone}
+        verb={verb}
+        textBeforeSendTime={textBeforeSendTime}
+        minutesOnHourPicker={minutesOnHourPicker}
+      />
+    </Box>
+  );
+};
+
+const ScheduleBody = ({
+  schedule,
+  updateSchedule,
+  scheduleOptions,
+  timezone,
+  verb,
+  textBeforeSendTime,
+  minutesOnHourPicker,
+}: Omit<ScheduleProps, "onScheduleChange"> & {
+  updateSchedule: UpdateSchedule;
+}) => {
+  const itemProps = {
+    schedule,
+    updateSchedule,
+  };
+
+  const Frequency = (
+    <ScheduleTypeSelect
+      key="frequency"
+      {...itemProps}
+      scheduleType={schedule.schedule_type}
+      scheduleOptions={scheduleOptions}
+    />
+  );
+  const Hour = (
+    <SelectTime
+      key="hour"
+      schedule={schedule}
+      updateSchedule={updateSchedule}
+    />
+  );
+
+  const TimeDetails = () => {
+    // eslint-disable-next-line react/prop-types
+    schedule.schedule_hour ??= null;
+    // eslint-disable-next-line react/prop-types
+    if (schedule.schedule_hour === null) {
+      return null;
+    }
+    if (!timezone) {
+      return null;
+    }
+    return (
+      <DisplayTimeDetails
+        key="time-details"
+        // eslint-disable-next-line react/prop-types
+        hour={schedule.schedule_hour}
+        // eslint-disable-next-line react/prop-types
+        amPm={schedule.schedule_hour >= 12 ? 1 : 0}
+        timezone={timezone}
+        textBeforeSendTime={textBeforeSendTime}
+      />
+    );
+  };
```

Comment:
declaring a component in a component like this isn't considered very React-y

Reply:
Let's chat next week - maybe I will have a chance to refactor this down the road

Reply:
turns out I had to refactor this right away to please the i18n checker :o)